identity/backend/src/main/java/io/camunda/identity/usermanagement/service/UserService.java

```diff
@@ -70,23 +80,30 @@ public List<CamundaUser> findAllUsers() {
     return userRepository.loadUsers();
   }
 
-  public CamundaUser updateUser(final String username, final CamundaUserWithPassword user) {
+  @Transactional
+  public CamundaUser updateUser(final Integer id, final CamundaUserWithPassword user) {
     try {
-      if (!username.equals(user.user().username())) {
+      if (!Objects.equals(id, user.user().id())) {
```

Comment:
❓Do we need to accept an ID here as well as a object that already has the ID? This can be simplified by just using `user.user().id()

Reply:
The need for adding id is derived from the fact that in api we are passing id as a path variable for update (PUT), so we somehow need to check the logic somewhere. I preferred it to be checked in service level.